pkg/cloud/metadata/ack_cluster_profile.go

```diff
@@ -27,7 +27,9 @@ func NewProfileMetadata(client kubernetes.Interface) (*ProfileMetadata, error) {
 
 func (m *ProfileMetadata) Get(key MetadataKey) (string, error) {
 	if key, ok := MetadataProfileDataKeys[key]; ok {
-		return m.profile.Data[key], nil
+		if m.profile.Data[key] != "" {
```

Comment:
It's unreasonable to return ErrUnknownMetadataKey for an empty value; please use a new error code instead.

Reply:
I think this is appropriate. Golang returns empty string if the key does not exist in the map. In this case,  ErrUnknownMetadataKey will trigger fallback to another source, which is expected.

Reply:
Just "unknown" for this source as keys are not found in cluster-profile.